### PR TITLE
fixes for -Wextra -Wpedantic -Wshadow -Wconversion

### PR DESCRIPTION
**Pull request by junghans**
Part of https://github.com/votca/votca/pull/212

### CODE REVIEW

File: include/votca/xtp/aotransform.h [hunk 1]
```diff
@@ -23,6 +23,10 @@
 
 #include <votca/xtp/aoshell.h>
 #include <votca/xtp/eigen.h>
+#if defined(__clang__)
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshadow"
```
Comment (junghans) [inline include/votca/xtp/aotransform.h]: @JensWehner that is for you to fix ;-)

Comment (JensWehner) [reply include/votca/xtp/aotransform.h]: okay? What happened there?

Comment (junghans) [reply include/votca/xtp/aotransform.h]: Just remove the block and compile with `-Wshadow` and have look at he warnings.

Comment (junghans) [reply include/votca/xtp/aotransform.h]: It might be fixed with `enum class`.

**Pull request merged**